notebook/services/contents/fileio.py

```diff
@@ -99,14 +99,8 @@ def atomic_writing(path, text=True, encoding='utf-8', log=None, **kwargs):
     if os.path.isfile(path):
         copy2_safe(path, tmp_path, log=log)
 
-    if text:
-        # Make sure that text files have Unix linefeeds by default
-        kwargs.setdefault('newline', '\n')
-        fileobj = io.open(path, 'w', encoding=encoding, **kwargs)
-    else:
-        fileobj = io.open(path, 'wb', **kwargs)
-
     try:
+        fileobj = open_file(path, text, encoding, log, **kwargs)
```

Comment:
To preserve the original integrity, this should occur prior to the `try` block, otherwise you need to protect the `fileobj.close()` call since `fileobj` will be `None` if `open_file()` throws an exception.